xclim/sdba/detrending.py

```diff
@@ -94,6 +141,11 @@ def _get_trend(self, grpd, dim="time"):
     def _fit(self, da):
         raise NotImplementedError
 
+    def __del__(self):
+        # Delete the training data file if it was saved to a temporary file.
+        if self.__ds_is_tempfile and hasattr(self, "_ds_file"):
+            self._ds_file.unlink()
```

Comment:
Should the file be `closed` ?

Reply:
I believe datasets are closed when garbage-collected? Will check.